src/System.Device.Gpio/System/Device/Gpio/Drivers/RaspberryPi3LinuxDriver.cs

```diff
@@ -113,9 +115,14 @@ protected internal override void AddCallbackForPinValueChangedEvent(int pinNumbe
         {
             ValidatePinNumber(pinNumber);
 
-            _interruptDriver.OpenPin(pinNumber);
-            _pinModes[pinNumber].InUseByInterruptDriver = true;
+            _interruptDriver!.OpenPin(pinNumber);

```

Comment:
Now that I see this, I wonder if this is actually accurate. Seems like this will only be initialized if we called `Initialize` method which would have only happened if we called `OpenPin` which I'm not 100% sure if that will always be the case when calling `AddCallbackForPinValueChangedEvent`

Reply:
Actually nevermind, seems like we are already protected against that because the GpioController will only let you add a callback if you have also already opened the pin, so I think we are good.

Reply:
I also tried to track that back. The way that `_interruptDriver` is set and used was a bit confusing to me. I came to the conclusion that this `!` was sound, but it wasn't immediately obvious.